src/DynamoCore/Nodes/NodeCategories.cs

```diff
@@ -430,28 +430,150 @@ internal static string MakeAbsolutePath(string basePath, string relativePath)
+        internal static IEnumerable<string> WrapText(string text, int maxCharacters)
         {
-            if (string.IsNullOrWhiteSpace(original))
-                return "";
-            StringBuilder newText = new StringBuilder(original.Length * 2);
-            newText.Append(original[0]);
-            for (int i = 1; i < original.Length; i++)
+            List<string> words = new List<string>();
+            if (string.IsNullOrWhiteSpace(text) || maxCharacters < 2)
+                return words;
+
+            StringBuilder currentWord = new StringBuilder(text[0].ToString());
+            for (int i = 1; i < text.Length; i++)
+            {
+                var curr = text[i];
+                var prev = text[i - 1];
+
+                if (Char.IsUpper(curr) && !Char.IsUpper(prev))
+                {
+                    words.Add(currentWord.ToString());
+                    currentWord.Clear();
+                }
+                currentWord.Append(curr);
+            }
+            // Add last word.
+            if (currentWord.Length != 0)
+                words.Add(currentWord.ToString());
+
+            // Try to merge words.
+            List<string> result_words = new List<string>();
+            currentWord.Clear();
+            currentWord.Append(words.First());
+            for (int i = 1; i < words.Count; i++)
             {
-                // We also have to check was previous character capital letter, e.g. Import From CSV                
-                var curr = original[i];
-                var prev = original[i - 1];
-                if ((Char.IsUpper(curr) || curr.Equals('(')) &&
-                    ((prev != ' ') && (!Char.IsUpper(prev))))
+                var word = words.ElementAt(i);
+                if ((currentWord.Length + word.Length + 1) <= maxCharacters)
                 {
-                    newText.Append(" ");
+                    currentWord.Append(" ");
+                    currentWord.Append(word);
                 }
-                newText.Append(original[i]);
+                else
+                {
+                    result_words.Add(currentWord.ToString());
+                    currentWord.Clear();
+                    currentWord.Append(word);
+                }
+            }
+            // Add last word.
+            if (currentWord.Length != 0)
+                result_words.Add(currentWord.ToString());
+
+            return result_words;
+        }
+
+        /// <summary>
+        /// Reduces the number of rows, based on the entries inside rows parameter.
+        /// E.g. rows = { "Insert", "Day", "Of", "Week", "Here" }, maxRows == 3
+        /// Result { "Insert", "Day", "Of Week Here" }
+        /// </summary>
+        /// <param name="rows">Incoming rows</param>
+        /// <param name="maxRows">Max number of rows</param>
+        internal static IEnumerable<string> ReduceRowCount(List<string> rows, int maxRows)
```

Comment:
Maybe we won't need to handle the special case of `rows.Count() <= maxRows`, can you try this?

```csharp
var results = new List<string>();
foreach (var row in rows)
{
    // There are still room in the results list.
    if (results.Length < maxRows)
    {
        results.Add(row);
        continue;
    }
    
    // Already full, keep appending to last row.
    var lastRow = results.Last();
    results.RemoveLast();
    results.Add(lastRow + " " + row);
}
```